File: leetcode/cpp/28_index_of_first_string.cpp
#include <iostream>
#include <string>
using namespace std;
 
int str_str(string haystack, string needle) {

    int needle_len = needle.length();
    int haystack_len = haystack.length();

    for (int i = 0; i < haystack_len - needle_len; i++) {

        for (int j = 0; j < needle_len; j++) {
            cout << haystack[i+j] << " " << needle[j] << endl;
            if (haystack[i+j] != needle[j]) {
                cout << "not found" << endl;
                break;
            }
            else {
                cout << "found" << endl;
            }
            cout << i << " " << j << endl;
        }
        
    }
}

int main() {
    bool result = str_str("happenstance", "sta");
    cout << "Result: " << result << endl;
}

File: leetcode/rust/28_index_of_first_string.rs

struct Solution {
    temp: i32,
}

impl Solution {
    pub fn str_str(haystack: String, needle: String) -> i32 {
        haystack.find(&needle).map(|i| i as i32).unwrap_or(-1)
    }
}

fn main() {
    println!("{}", Solution::str_str("yepasdf".to_string(), "asdf".to_string()));
}